### DISCUSSION TITLE
Ejecting

### DISCUSSION THREAD

**Discussion by goldenpath-ai** [directus/directus#15827] [Feature Requests]
Hello,

what do you think about giving users the option to eject and have the ability to modify everything? Right now developing custom directus extensions can be quite a pain in more complex scenarios since your hands get tied up pretty quickly. 

Having the ability to eject and directly modify all of the files and configs would be of great help. 

Thank you for considering.

**Comment by nickrum** [comment]
Could you elaborate on what exactly you mean by "ejecting" and "the ability to modify everything"? I'm assuming you are referring to [this](https://github.com/facebook/create-react-app/blob/main/packages/cra-template/template/README.md#npm-run-eject)?

This would be very tricky, because the operations performed by the Extensions SDK are more complex than those performed by CRA.

Where exactly were your hands getting tied up? The Extensions SDK already has support for a config file which we can use to further customize the behavior during the build process.

**Reply by goldenpath-ai** [reply]
Yes, I mean ejecting in the same way as done in cra. 

For example I cannot use libraries that require me to use Vue.use(). Which is probably the biggest issue here. There might be some workarounds but I haven't been able to find any. Either exposing the index Vue file or documenting a workaround would be great. Also libraries that are referencing their non-js files from a folder inside the library's directory don't work because of how Directus handles minification in custom modules. (Puts everything in one file in a folder so relative imports in the library code break)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
